### PR TITLE
Timeout channels on failed session setup

### PR DESCRIPTION
**Pull request by zolia**
Fixes: #2200

easily reporducible by attempting to connect to your node from client on the same network.﻿Should fix port mappings piling up on routers.

### CODE REVIEW

File: p2p/channel.go [hunk 1]
```diff
@@ -511,6 +519,24 @@ func (c *channel) setUpnpPortsRelease(release []func()) {
 	c.upnpPortsRelease = release
 }
 
+func (c *channel) checkIfChannelAlive() {
+	for {
+		select {
+		case <-c.stop:
+			return
+		case <-time.After(initialTrafficTimeout):
+			if c.remoteAlive {
```
Comment (vkuznecovas) [inline p2p/channel.go:528]: you're gonna race here. Use a channel and add a third case:

```
case <-c.remoteAlive:
   return
```

Comment (zolia) [reply p2p/channel.go]: With channel is cleaner anyways. Thanx.

File: p2p/channel.go [hunk 2]
```diff
@@ -244,6 +252,10 @@ func (c *channel) remoteReadLoop() {
 			return
 		}
 
+		c.once.Do(func() {
```
Comment (soffokl) [inline p2p/channel.go:255]: You cannot use `once` twice. It's already used in `Close()`.

Comment (zolia) [reply p2p/channel.go:255]: You can and it works, since you are performing different action with it.

Comment (soffokl) [reply p2p/channel.go:255]: https://play.golang.org/p/Gyyqn8EwQpI

Comment (vkuznecovas) [reply p2p/channel.go:255]: confirming it does not work.

File: p2p/channel.go [hunk 3]
```diff
@@ -511,6 +523,22 @@ func (c *channel) setUpnpPortsRelease(release []func()) {
 	c.upnpPortsRelease = release
 }
 
+func (c *channel) checkIfChannelAlive() {
+	for {
```
Comment (anjmao) [inline p2p/channel.go:527]: for loop is not needed here as you want this logic to execute only once

File: p2p/channel.go [hunk 4]
```diff
@@ -236,6 +242,8 @@ func (c *channel) remoteReadLoop() {
 		default:
 		}
 
+		go c.checkIfChannelAlive()
```
Comment (anjmao) [inline p2p/channel.go:245]: checkIfChannelAlive can be moved outside for loop

File: p2p/channel.go (hunk 2 above)
Comment (zolia) [reply p2p/channel.go:255]: True, we figured out why it worked in my case. Actual channel close was not being executed after successful session.

**Pull request merged**